distributed/scheduler.py

```diff
@@ -3604,29 +3604,27 @@ async def close_worker(self, comm=None, worker=None, safe=None):
     def heartbeat_worker(
         self,
         comm=None,
-        address=None,
-        resolve_address=True,
-        now=None,
-        resources=None,
-        host_info=None,
-        metrics=None,
-        executing=None,
+        *,
+        address,
+        resolve_address: bool = True,
+        now: float = None,
+        resources: dict = None,
+        host_info: dict = None,
+        metrics: dict,
+        executing: dict = None,
```

Comment:
FWIW I find that `address` and `metrics` dropping default values here to be more confusing than what we had before. What does it mean for these parameters to not have a default but also be after the `*`?

Reply:
It means that they are mandatory keyword arguments.
Nothing in the code caters of address=None - if it arrives, it's not going to be dealt with correctly.
There was an explicit ``assert metrics``  which prevented you from omitting the parameter.

Reply:
Alright, thanks for clarifying -- this was a bit jarring to see for the first time